x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/register.ts

```diff
@@ -35,14 +35,18 @@ export function registerCspmUsageCollector(
       return true;
     },
     fetch: async (collectorFetchContext: CollectorFetchContext) => {
-      const [
-        indicesStats,
-        accountsStats,
-        resourcesStats,
-        rulesStats,
-        installationStats,
-        alertsStats,
-      ] = await Promise.all([
+      const handleResult = <T>(taskName: string, result: PromiseSettledResult<T>) => {
+        if (result.status === 'fulfilled') {
+          return result.value;
+        } else {
+          // Handle the error and log it
+          logger.error(`${taskName} task failed: ${result.reason}`);
+
+          // Return null or some default value, depending on your needs
+          return [];
```

Comment:
When there's an error it makes more sense to return `undefined` (meaning not returning any value)

it is more common to use that as an empty state of issue than any other value